packages/gatsby/src/redux/actions/public.ts

```diff
@@ -395,14 +386,14 @@ ${reservedFields.map(f => `  * "${f}"`).join(`\n`)}
     page.path = truncatedPath
   }
 
-  const internalPage: Page = {
+  const internalPage: IPage = {
     internalComponentName,
     path: page.path,
     matchPath: page.matchPath,
     component: page.component,
     componentChunkName: generateComponentChunkName(page.component),
     isCreatedByStatefulCreatePages:
-      actionOptions &&
+      !!actionOptions &&
       actionOptions.traceId === `initial-createPagesStatefully`,
```

Comment:
we can improve this: `actionOptions?.traceId === "initial-createPagesStatefully"`